custom_components/tahoma/coordinator.py

```diff
@@ -52,12 +53,15 @@ async def _async_update_data(self) -> Dict[str, Device]:
                 if event.name == "DeviceStateChangedEvent":
                     for state in event.device_states:
                         device = self.devices[event.deviceurl]
-                        device.states[state.name].value = self._get_state(state)
+                        if state.name not in device.states:
+                            device.states[state.name] = state
+                        else:
```

Comment:
You can remove the else. This way with the next line we will cast the state value.